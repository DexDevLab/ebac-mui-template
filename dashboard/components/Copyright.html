<!DOCTYPE html><html lang="pt-br"><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><meta name="next-head-count" content="2"/><link rel="preload" href="./_next/static/css/876d048b5dab7c28.css" as="style"/><link rel="stylesheet" href="./_next/static/css/876d048b5dab7c28.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="./_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="./_next/static/chunks/webpack-5f3bdadbaf77b361.js" defer=""></script><script src="./_next/static/chunks/framework-2c79e2a64abdb08b.js" defer=""></script><script src="./_next/static/chunks/main-9a78aff16b7f6647.js" defer=""></script><script src="./_next/static/chunks/pages/_app-0329340c8e72f1ad.js" defer=""></script><script src="./_next/static/chunks/861-de8856a012482621.js" defer=""></script><script src="./_next/static/chunks/664-abc38ceedcbded10.js" defer=""></script><script src="./_next/static/chunks/pages/dashboard/components/Copyright-a62860af1bd6a7bf.js" defer=""></script><script src="./_next/static/EjffxaP_CM_-lGCD--LWB/_buildManifest.js" defer=""></script><script src="./_next/static/EjffxaP_CM_-lGCD--LWB/_ssgManifest.js" defer=""></script></head><body><div id="__next"><style data-emotion="css-global 11bg8zp">html{-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;box-sizing:border-box;-webkit-text-size-adjust:100%;}*,*::before,*::after{box-sizing:inherit;}strong,b{font-weight:700;}body{margin:0;color:rgba(251,250,253,0.87);font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;background-color:#000000;}@media print{body{background-color:#fff;}}body::backdrop{background-color:#000000;}</style><style data-emotion="css 1wb0ge5">.css-1wb0ge5{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;text-align:center;color:rgba(251,250,253,0.87);}</style><p class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1wb0ge5">Copyright Â© <a color="inherit" href="/dashboard/components/Copyright#">Daniel Almeida</a> <!-- -->2023<!-- -->.</p></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/dashboard/components/Copyright","query":{},"buildId":"EjffxaP_CM_-lGCD--LWB","assetPrefix":".","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}</script></body></html>