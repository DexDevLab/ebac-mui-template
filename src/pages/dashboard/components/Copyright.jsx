import { Typography } from "@mui/material";
import Link from "next/link";
import * as React from "react";

export default function Copyright({page, ...props}) {

    const STATUS = {
      HOVERED: "hovered",
      NORMAL: "normal",
    };
  
    const [status, setStatus] = React.useState(STATUS.NORMAL);
  
    const onMouseEnter = () => {
      setStatus(STATUS.HOVERED);
    };
  
    const onMouseLeave = () => {
      setStatus(STATUS.NORMAL);
    };
    return (
      <Typography
        variant="body2"
        color="text.secondary"
        align="center"
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        {...props}
      >
        {"Copyright Â© "}
        <Link
          color="inherit"
          href={page || '#'}
        //   href="https://github.com/DexDevLab"
        >
          Daniel Almeida
        </Link>{" "}
        {new Date().getFullYear()}
        {"."}
      </Typography>
    );
  }