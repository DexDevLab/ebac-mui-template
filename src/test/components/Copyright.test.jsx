/**
 * @jest-environment jsdom
 */

import renderer from "react-test-renderer";
import Copyright from "../../pages/dashboard/components/Copyright";

it("changes the class when hovered", () => {
  const component = renderer.create(
    <Copyright page="https://github.com/DexDevLab">Daniel Almeida</Copyright>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() => {
    tree.props.onMouseLeave();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
