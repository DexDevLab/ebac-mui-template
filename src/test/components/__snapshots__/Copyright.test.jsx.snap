// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Copyright © 
  <a
    color="inherit"
    href="https://github.com/DexDevLab"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Daniel Almeida
  </a>
   
  2023
  .
</p>
`;

exports[`changes the class when hovered 2`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Copyright © 
  <a
    color="inherit"
    href="https://github.com/DexDevLab"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Daniel Almeida
  </a>
   
  2023
  .
</p>
`;

exports[`changes the class when hovered 3`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Copyright © 
  <a
    color="inherit"
    href="https://github.com/DexDevLab"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Daniel Almeida
  </a>
   
  2023
  .
</p>
`;
